### PR TITLE
Integrate listing and deleting changelogs in admin panel

### PR DESCRIPTION
**Pull request by rohithbalaji123**
Part of #612 

### Description
Integrate displaying the list of change logs in the admin panel and deleting each changelog in the changelogs tab.

Few things that changed from the design [here](https://app.zeplin.io/project/5ee6fa9f0d418aba36650713/screen/5ee6fbac342247ba1fc8acfd) are,
1. Title of the table headers. The table headers are by default made to be capitalized throughout the project. Changing that will required more support for customizations in the table component.
1. Date format in Released At column. The format given in the design is a bit confusing as both dd/mm/yyyy and mm/dd/yyyy are commonly used and also, it doesn't specify the timezone.

### Screenshots
image

### CODE REVIEW

File: frontend/src/component/pages/admin/changelog/ManageChangeLogsSection.tsx [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    if (this.isChangeRowExpanded(changeId)) {
+      this.collapseChangeRow();
+      return;
+    }
+    this.expandChangeRow(changeId);
+  };
+
+  private isChangeRowExpanded(changeId: string) {
+    return this.state.expandedChangeId === changeId;
+  }
```
Comment (byliuyang) [inline frontend/src/component/pages/admin/changelog/ManageChangeLogsSection.tsx:80]: Multiple rows can be expanded at the same time. I think we need to keep a map between changeID and whether they are expanded.

File: frontend/src/component/ui/Icon.tsx [hunk 2]
```diff
@@ -48,6 +51,12 @@ export class Icon extends Component<IProps> {
         return this.renderEditIcon();
       case IconID.Check:
         return this.renderCheckIcon();
+      case IconID.Delete:
+        return this.renderDeleteIcon();
+      case IconID.RightArrow:
+        return this.renderRightArrowIcon();
+      case IconID.DownArrow:
+        return this.renderDownArrowIcon();
```
Comment (byliuyang) [inline frontend/src/component/ui/Icon.tsx:59]: Not needed with CSS transform.

File: frontend/src/component/ui/Icon.tsx [hunk 3]
```diff
@@ -8,7 +8,10 @@ export enum IconID {
   Close,
   Search,
   Edit,
-  Check
+  Check,
+  Delete,
+  RightArrow,
+  DownArrow
```
Comment (byliuyang) [inline frontend/src/component/ui/Icon.tsx:14]:
```suggestion
```

File: frontend/src/component/ui/Icon.tsx [hunk 4]
```diff
@@ -120,4 +129,48 @@ export class Icon extends Component<IProps> {
...
+  private renderDownArrowIcon = () => {
+    return (
+      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24">
+        <path
+          d="M 23.707031 5.792969 C 23.316406 5.402344 22.683594 5.402344 22.292969 5.792969 L 12 16.085938 L 1.707031 5.792969 C 1.316406 5.402344 0.683594 5.402344 0.292969 5.792969 C -0.0976562 6.183594 -0.0976562 6.816406 0.292969 7.207031 L 11.292969 18.207031 C 11.488281 18.402344 11.742188 18.5 12 18.5 C 12.257812 18.5 12.511719 18.402344 12.707031 18.207031 L 23.707031 7.207031 C 24.097656 6.816406 24.097656 6.183594 23.707031 5.792969 Z M 23.707031 5.792969 "
+          fill="#000000"
+        />
+      </svg>
+    );
+  };
```
Comment (byliuyang) [inline frontend/src/component/ui/Icon.tsx:175]:
```suggestion
```

File: frontend/src/state/actions.ts [hunk 5]
```diff
@@ -6,11 +6,16 @@ export interface IPayloadAction extends Action {
...
 export const RAISE_GET_USER_SHORT_LINKS_ERROR =
   'RAISE_GET_USER_SHORT_LINKS_ERROR';
+
+export const CLEAR_ERROR = 'CLEAR_ERROR';
+
 export const RAISE_GET_CHANGELOG_ERROR = 'RAISE_GET_CHANGELOG_ERROR';
+
 export const RAISE_CREATE_CHANGE_ERROR = 'RAISE_CREATE_CHANGE_ERROR';
+export const RAISE_DELETE_CHANGE_ERROR = 'RAISE_DELETE_CHANGE_ERROR';
+export const RAISE_GET_ALL_CHANGES_ERROR = 'RAISE_GET_ALL_CHANGEs_ERROR';
```
Comment (byliuyang) [inline frontend/src/state/actions.ts:18]:
```suggestion
export const RAISE_GET_ALL_CHANGES_ERROR = 'RAISE_GET_ALL_CHANGES_ERROR';
```

**Comment by byliuyang** [comment]
Hi @rohithbalaji123 , thank you so much for working on this cool feature! I left few comments on the PR!

File: frontend/src/component/pages/admin/changelog/ChangeLogTab.tsx [hunk 6]
```diff
@@ -13,9 +19,25 @@ interface IProps {
...
+  constructor(props: IProps) {
+    super(props);
+
+    this.state = {
+      changes: []
+    };
+  }
+
+  componentDidMount() {
+    this.getChanges();
```
[the remark this answers is not in the document]
Comment (rohithbalaji123) [reply frontend/src/component/pages/admin/changelog/ChangeLogTab.tsx:38]: @byliuyang , that's not the recommended way as there is a possibility that `setState` can be called even before the component is rendered. Moreover the React docs recommend to make any API calls to fetch data in `componentDidMount`.
https://reactjs.org/docs/faq-ajax.html#where-in-the-component-lifecycle-should-i-make-an-ajax-call

**Pull request merged**